lib/mix/lib/mix/tasks/test.ex

```diff
@@ -692,30 +690,19 @@ defmodule Mix.Tasks.Test do
     [autorun: false] ++ opts
   end
 
-  defp parse_files([], _shell, test_paths) do
+  defp parse_file_paths([], test_paths) do
     test_paths
   end
 
-  defp parse_files([single_file], _shell, _test_paths) do
-    # Check if the single file path matches test/path/to_test.exs:123. If it does,
-    # apply "--only line:123" and trim the trailing :123 part.
-    {single_file, opts} = ExUnit.Filters.parse_path(single_file)
-    ExUnit.configure(opts)
-    [single_file]
-  end
-
-  defp parse_files(files, shell, _test_paths) do
-    if Enum.any?(files, &match?({_, [_ | _]}, ExUnit.Filters.parse_path(&1))) do
-      raise_with_shell(shell, "Line numbers can only be used when running a single test file")
-    else
-      files
-    end
+  defp parse_file_paths(file_paths, _test_paths) do
+    {parsed_file_paths, ex_unit_opts} = ExUnit.Filters.parse_paths(file_paths)
+    ExUnit.configure(ex_unit_opts)
+    parsed_file_paths
   end
 
   defp parse_filters(opts, key) do
-    if Keyword.has_key?(opts, key) do
-      ExUnit.Filters.parse(Keyword.get_values(opts, key))
-    end
+    values = Keyword.get_values(opts, key)
+    if values != [], do: ExUnit.Filters.parse(values)
```

Comment:
Please keep the previous version, I thinking looking up a key is faster than trying to accumulate all upfront. Also, this has not changed in this PR, so we should probably keep it the same. :)

Reply:
Yeah, sorry. Felt tempted. :sweat_smile: